lib/Tools/LinearLayout.cpp

```diff
@@ -989,6 +953,30 @@ LinearLayout::getFreeVariableMasks() const {
   return ret;
 }
 
+size_t hash_value(const LinearLayout &layout) {
```

Comment:
Where do you use this function?

Reply:
It's used internally by MLIR when using `LinearLayout` as an attribute.